In [24]:
import pandas as pd
import numpy as np

In [3]:
data = pd.read_csv("/content/FoodBalanceSheets_E_Africa_NOFLAG.csv", encoding = 'latin-1')

In [5]:
data.head()

Area Code     Area  Item Code         Item  Element Code  \
0          4  Algeria       2501   Population           511   
1          4  Algeria       2501   Population          5301   
2          4  Algeria       2901  Grand Total           664   
3          4  Algeria       2901  Grand Total           674   
4          4  Algeria       2901  Grand Total           684   

                                  Element             Unit     Y2014  \
0           Total Population - Both sexes     1000 persons  38924.00   
1                Domestic supply quantity      1000 tonnes      0.00   
2           Food supply (kcal/capita/day)  kcal/capita/day   3377.00   
3  Protein supply quantity (g/capita/day)     g/capita/day     94.90   
4      Fat supply quantity (g/capita/day)     g/capita/day     80.06   

      Y2015     Y2016     Y2017     Y2018  
0  39728.00  40551.00  41389.00  42228.00  
1      0.00      0.00      0.00      0.00  
2   3379.00   3372.00   3341.00   3322.00  
3     94.35     94.72     92.82     91.83  
4     79.36     77.40     80.19     77.28

In [9]:
len(data.Area.unique())

49

In [6]:
area_df = data.groupby('Area')['Y2017'].sum().reset_index()

In [12]:
area_df.head()

Area      Y2017
0       Algeria  325644.27
1        Angola  229159.57
2         Benin  124771.22
3      Botswana   22101.30
4  Burkina Faso  101855.07

In [35]:
area_df.sort_values('Y2017', ascending=False).head(1)

Area       Y2017
34  Nigeria  1483268.23

The Area that had the highest sum in 2017 was Nigeria

In [16]:
correlation_data = data[['Element Code', 'Y2014', 'Y2015', 'Y2016', 'Y2017', 'Y2018']]

In [17]:
correlation_data.head()

Element Code     Y2014     Y2015     Y2016     Y2017     Y2018
0           511  38924.00  39728.00  40551.00  41389.00  42228.00
1          5301      0.00      0.00      0.00      0.00      0.00
2           664   3377.00   3379.00   3372.00   3341.00   3322.00
3           674     94.90     94.35     94.72     92.82     91.83
4           684     80.06     79.36     77.40     80.19     77.28

In [18]:
correlation_data.corr()

Element Code     Y2014     Y2015     Y2016     Y2017     Y2018
Element Code      1.000000  0.024457  0.023889  0.023444  0.024254  0.024279
Y2014             0.024457  1.000000  0.994647  0.996081  0.995230  0.994872
Y2015             0.023889  0.994647  1.000000  0.995739  0.988048  0.988208
Y2016             0.023444  0.996081  0.995739  1.000000  0.992785  0.992757
Y2017             0.024254  0.995230  0.988048  0.992785  1.000000  0.998103
Y2018             0.024279  0.994872  0.988208  0.992757  0.998103  1.000000

The year that had the least correlation with ‘Element Code’ is 2016 from the above table

In [20]:
data.Area.unique()

array(['Algeria', 'Angola', 'Benin', 'Botswana', 'Burkina Faso',
       'Cabo Verde', 'Cameroon', 'Central African Republic', 'Chad',
       'Comoros', 'Congo', "Côte d'Ivoire", 'Djibouti', 'Egypt',
       'Eswatini', 'Ethiopia', 'Ethiopia PDR', 'Gabon', 'Gambia', 'Ghana',
       'Guinea', 'Guinea-Bissau', 'Kenya', 'Lesotho', 'Liberia',
       'Madagascar', 'Malawi', 'Mali', 'Mauritania', 'Mauritius',
       'Morocco', 'Mozambique', 'Namibia', 'Niger', 'Nigeria', 'Rwanda',
       'Sao Tome and Principe', 'Senegal', 'Seychelles', 'Sierra Leone',
       'South Africa', 'Sudan', 'Sudan (former)', 'Togo', 'Tunisia',
       'Uganda', 'United Republic of Tanzania', 'Zambia', 'Zimbabwe'],
      dtype=object)

In [26]:
data.Element.unique()

array(['Total Population - Both sexes', 'Domestic supply quantity',
       'Food supply (kcal/capita/day)',
       'Protein supply quantity (g/capita/day)',
       'Fat supply quantity (g/capita/day)', 'Production',
       'Import Quantity', 'Stock Variation', 'Export Quantity', 'Feed',
       'Seed', 'Losses', 'Processing', 'Other uses (non-food)',
       'Residuals', 'Food', 'Food supply quantity (kg/capita/yr)',
       'Tourist consumption'], dtype=object)

In [29]:
madagascar_df = data[data.Area == 'Madagascar']

In [32]:
madagascar_protien = madagascar_df[madagascar_df.Element == 'Protein supply quantity (g/capita/day)']

In [34]:
np.sum(madagascar_protien.Y2015)

173.04999999999998

The total Protein supply quantity in Madagascar in 2015 was 173.05

In [42]:
area_df.sort_values('Y2017', ascending=False).tail(8)

Area     Y2017
23                Lesotho  21267.96
21          Guinea-Bissau  19102.77
5              Cabo Verde  14650.74
36  Sao Tome and Principe  12662.63
38             Seychelles    442.34
9                 Comoros     59.84
42         Sudan (former)      0.00
16           Ethiopia PDR      0.00

The Area that had the 7th lowest sum in 2017 was Guinea-Bissau

In [49]:
data.isnull().sum()

Area Code          0
Area               0
Item Code          0
Item               0
Element Code       0
Element            0
Unit               0
Y2014           1589
Y2015           1548
Y2016           1535
Y2017           1506
Y2018           1436
dtype: int64

In [52]:
data.shape[0]

60943

In [53]:
percentage = (1589 / data.shape[0]) * 100
percentage

2.6073544131401474


The total number and percentage of missing data in 2014 was 1589 and 2.61%

In [57]:
item_df = data.groupby('Item')[['Y2015', 'Y2018']].sum().reset_index()

In [62]:
item_df[item_df['Item'] == "Wine"]

Item    Y2015    Y2018
117  Wine  4251.81  4039.32

The total sum of Wine produced in 2015 and 2018 respectively were 4251.81 and 4039.32

In [64]:
len(data.Area.unique())

49

The total number of unique countries in the dataset is 49 countries.

In [65]:
element_df = data.groupby('Element')[['Y2014', 'Y2015', 'Y2016', 'Y2017', 'Y2018']].sum().reset_index()

In [66]:
element_df

Element       Y2014       Y2015  \
0                 Domestic supply quantity  1996716.35  2021493.55   
1                          Export Quantity   150020.64   157614.47   
2       Fat supply quantity (g/capita/day)    10225.56    10235.74   
3                                     Feed   216927.89   225050.22   
4                                     Food  1212332.49  1232361.10   
5            Food supply (kcal/capita/day)   454257.00   453383.00   
6      Food supply quantity (kg/capita/yr)    49650.63    49345.13   
7                          Import Quantity   274144.48   267018.46   
8                                   Losses   153223.00   155439.00   
9                    Other uses (non-food)    78718.13    66254.41   
10                              Processing   282923.00   287929.00   
11                              Production  1931287.75  1947019.39   
12  Protein supply quantity (g/capita/day)    11836.46    11833.95   
13                               Residuals    30149.00    30045.00   
14                                    Seed    21922.92    23976.82   
15                         Stock Variation    58749.83    34910.99   
16           Total Population - Both sexes  1031585.00  1058081.00   
17                     Tourist consumption      416.00      349.00   

         Y2016       Y2017       Y2018  
0   2044842.70  2088198.10  2161192.10  
1    151920.46   182338.80   181594.80  
2     10102.77    10253.84    10258.69  
3    228958.65   223705.68   233489.68  
4   1247022.17  1258888.28  1303841.28  
5    451810.00   454681.00   455261.00  
6     48985.28    48690.04    49056.85  
7    286582.78   294559.09   287997.09  
8    157787.00   160614.00   163902.00  
9     69563.68    91645.97    91300.97  
10   280631.00   292836.00   308429.00  
11  1943537.15  2030056.89  2075072.89  
12    11779.69    11842.45    11833.56  
13    37224.00    35500.00    34864.00  
14    23389.20    24870.14    25263.14  
15    33140.12    54316.91    20577.91  
16  1085107.00  1112641.00  1140605.00  
17       89.00       91.00       90.00

The year with the highest sum of Stock Variation is 2014.

In [67]:
data.describe()

Area Code     Item Code  Element Code          Y2014          Y2015  \
count  60943.000000  60943.000000  60943.000000   59354.000000   59395.000000   
mean     134.265576   2687.176706   3814.856456     134.196282     135.235966   
std       72.605709    146.055739   2212.007033    1567.663696    1603.403984   
min        4.000000   2501.000000    511.000000   -1796.000000   -3161.000000   
25%       74.000000   2562.000000    684.000000       0.000000       0.000000   
50%      136.000000   2630.000000   5142.000000       0.090000       0.080000   
75%      195.000000   2775.000000   5511.000000       8.340000       8.460000   
max      276.000000   2961.000000   5911.000000  176405.000000  181137.000000   

               Y2016          Y2017          Y2018  
count   59408.000000   59437.000000   59507.000000  
mean      136.555222     140.917765     143.758381  
std      1640.007194    1671.862359    1710.782658  
min     -3225.000000   -1582.000000   -3396.000000  
25%         0.000000       0.000000       0.000000  
50%         0.080000       0.100000       0.070000  
75%         8.430000       9.000000       9.000000  
max    185960.000000  190873.000000  195875.000000

The mean and standard deviation across the whole dataset for the year 2017 were 140.92, 1671.86 respectively.

In [69]:
element_df

Element       Y2014       Y2015  \
0                 Domestic supply quantity  1996716.35  2021493.55   
1                          Export Quantity   150020.64   157614.47   
2       Fat supply quantity (g/capita/day)    10225.56    10235.74   
3                                     Feed   216927.89   225050.22   
4                                     Food  1212332.49  1232361.10   
5            Food supply (kcal/capita/day)   454257.00   453383.00   
6      Food supply quantity (kg/capita/yr)    49650.63    49345.13   
7                          Import Quantity   274144.48   267018.46   
8                                   Losses   153223.00   155439.00   
9                    Other uses (non-food)    78718.13    66254.41   
10                              Processing   282923.00   287929.00   
11                              Production  1931287.75  1947019.39   
12  Protein supply quantity (g/capita/day)    11836.46    11833.95   
13                               Residuals    30149.00    30045.00   
14                                    Seed    21922.92    23976.82   
15                         Stock Variation    58749.83    34910.99   
16           Total Population - Both sexes  1031585.00  1058081.00   
17                     Tourist consumption      416.00      349.00   

         Y2016       Y2017       Y2018  
0   2044842.70  2088198.10  2161192.10  
1    151920.46   182338.80   181594.80  
2     10102.77    10253.84    10258.69  
3    228958.65   223705.68   233489.68  
4   1247022.17  1258888.28  1303841.28  
5    451810.00   454681.00   455261.00  
6     48985.28    48690.04    49056.85  
7    286582.78   294559.09   287997.09  
8    157787.00   160614.00   163902.00  
9     69563.68    91645.97    91300.97  
10   280631.00   292836.00   308429.00  
11  1943537.15  2030056.89  2075072.89  
12    11779.69    11842.45    11833.56  
13    37224.00    35500.00    34864.00  
14    23389.20    24870.14    25263.14  
15    33140.12    54316.91    20577.91  
16  1085107.00  1112641.00  1140605.00  
17       89.00       91.00       90.00

The total number of the sum of Processing in 2017 was 292836.00.